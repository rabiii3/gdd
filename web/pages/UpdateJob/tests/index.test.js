import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import configureStore from '../../../store/configureStore.test';
import { enhance } from '..';

describe( 'app | pages | UpdateJobDescription | index', () => {
  it('should match snapshot', () => {
    const Component = () => <div/>;
    const EnhancedComponent = enhance(Component);
    const state = {
      jobDescription:{
        data:{
          '0':{
            _id: '0',
            title: 'required js dev',
            content: 'some content',
            skills: ['5d035ef378c332001b67bada', '5d035ef378c332001b67badb'],
            createdBy: '456',
            createdAt: '',
            updatedBy: '456',
            updatedAt: '',
            tenantId: '333'
          }
        },
        sort:{
          sortBy: 'TITLE',
          direction: 'desc',
        }
      },
    };

    const store = configureStore(state);
    const wrapper = mount(
      <Provider store={store}>
        <EnhancedComponent match={{params: {id: 1}, isExact: true, path: "", url: ""}} />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});