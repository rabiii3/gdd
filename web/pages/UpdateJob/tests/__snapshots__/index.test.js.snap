// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app | pages | UpdateJobDescription | index should match snapshot 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <WithStyles(withStateHandlers(Connect(lifecycle(Component))))
    match={
      Object {
        "isExact": true,
        "params": Object {
          "id": 1,
        },
        "path": "",
        "url": "",
      }
    }
  >
    <withStateHandlers(Connect(lifecycle(Component)))
      classes={
        Object {
          "root": "withStateHandlers-Connect-lifecycle-Component----root-1",
        }
      }
      match={
        Object {
          "isExact": true,
          "params": Object {
            "id": 1,
          },
          "path": "",
          "url": "",
        }
      }
    >
      <Connect(lifecycle(Component))
        classes={
          Object {
            "root": "withStateHandlers-Connect-lifecycle-Component----root-1",
          }
        }
        isEditMode={false}
        match={
          Object {
            "isExact": true,
            "params": Object {
              "id": 1,
            },
            "path": "",
            "url": "",
          }
        }
        setEditMode={[Function]}
      >
        <lifecycle(Component)
          classes={
            Object {
              "root": "withStateHandlers-Connect-lifecycle-Component----root-1",
            }
          }
          dispatch={[Function]}
          isEditMode={false}
          match={
            Object {
              "isExact": true,
              "params": Object {
                "id": 1,
              },
              "path": "",
              "url": "",
            }
          }
          setEditMode={[Function]}
        >
          <Component
            classes={
              Object {
                "root": "withStateHandlers-Connect-lifecycle-Component----root-1",
              }
            }
            dispatch={[Function]}
            isEditMode={false}
            match={
              Object {
                "isExact": true,
                "params": Object {
                  "id": 1,
                },
                "path": "",
                "url": "",
              }
            }
            setEditMode={[Function]}
          >
            <div />
          </Component>
        </lifecycle(Component)>
      </Connect(lifecycle(Component))>
    </withStateHandlers(Connect(lifecycle(Component)))>
  </WithStyles(withStateHandlers(Connect(lifecycle(Component))))>
</Provider>
`;
